main.go

```diff
@@ -172,21 +152,41 @@ func main() {
 // 	Utility structs
 //
 
-type Mutex struct {
-	mu     sync.Mutex
-	locked bool
+type VideoRunner struct {
+	mx      sync.Mutex
+	running bool
 }
 
-func (m *Mutex) Lock() {
-	m.mu.Lock()
-	m.locked = true
-}
+func (t *VideoRunner) playVideo(vlcBinaryPath string, video string, display Display) bool {
+	t.mx.Lock()
+	if t.running {
+		t.mx.Unlock()
+		return false
+	}
+	t.running = true
+	t.mx.Unlock()
 
-func (m *Mutex) Unlock() {
-	m.locked = false
-	m.mu.Unlock()
-}
+	go func() {
```

Comment:
Kicking off in a goroutine because I want the request to return immediately still